### PR TITLE
add auth0 authentication

### PR DESCRIPTION
**Pull request by muhammadn**
This is the code to include the auth0 authentication.

You will need to run `rake db:seed` again to create the necessary data for Auth0 in the settings page.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -277,13 +279,14 @@ GEM
     omniauth-twitter (1.4.0)
       omniauth-oauth (~> 1.1)
       rack
+    parallel (1.12.0)
     parser (2.4.0.0)
       ast (~> 2.2)
-    pg (0.20.0)
+    pg (0.21.0)
```
Comment (thorsteneckel) [inline Gemfile.lock]: Please don't perform `bundle update` since some dependency versions (e.g. `pg`) are currently broken 😞

File: .rubocop.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ Style/LeadingCommentSpace:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#hash-space'
   Enabled: false
 
-Style/MethodCallParentheses:
+Style/MethodCallWithoutArgsParentheses:
```
Comment (thorsteneckel) [inline .rubocop.yml]: Changes to this file won't be necessary after the previous `rubocop` version is used.

File: Gemfile.lock (hunk 1 above)
Comment (muhammadn) [reply Gemfile.lock]: Okay, sorry about that. Do you want me to make another new PR instead?

File: .rubocop.yml (hunk 2 above)
Comment (muhammadn) [reply .rubocop.yml]: will remove that.

**Comment by muhammadn** [comment]
I have created an isolated temporary endpoint here (which i will remove after you have tested):

site: https://taqisystems.auth0.com
app id: ujG5FREzJFykqLo4Z9osd6ASal4qcIS4
secret: 4Eoex-JyaB1UJUX4-CziKX-ysRllzCxlLi564kLwBRTWhW0Iwf7r-K6yXMNsAxpS

username: test@zammad.com
password: password123

**Comment by thorsteneckel** [comment]
Hi @muhammadn - Sorry for the late response. I discussed this internally and we came to a similar conclusion as for another Pull Request I answered earlier. This is clearly a similar place where we need an interface to extend the functionality. Pull requests would be great for this!

Even if auth0 and OpenID are used occasionally, this strategy is not widely used. In addition, this means additional dependencies and maintenance that cannot be borne.

We must therefore refrain from this function in the standard system. However, an extension would be exactly the right application. For this, however, the extensibility described above must first be implemented.

For these reasons, I must reject the pull request.

I admire your great contributions to this project. I don't want to slow you down or get in your way. We will be happy to discuss in Issues you would like to implement whether it is the right time and how a solution can look like. Thank you!

**Comment by xeruf** [comment]
Any progress on the extensibility front?

Is this tracked anywhere?

**Pull request closed without merging**